crates/holochain/src/core/workflow/sys_validation_workflow.rs

```diff
@@ -1390,3 +1478,118 @@ impl Default for OutcomeSummary {
         OutcomeSummary::new()
     }
 }
+
+/// Test the detect_fork function against different situations,
+/// especially the case where a fork happens after an Update Agent action,
+/// where the authorship changes
+#[tokio::test(flavor = "multi_thread")]
+async fn test_detect_fork() {
```

Comment:
Is this test inside a `cfg(test)`? There are a bunch of test modules under sys validation, maybe this belongs in one of those, or maybe a new one?

Reply:
Any use of `#[test]` including `#[tokio::test]` is already `cfg(test)`. We do have a pattern of putting tests inside their own module though so I will do that. I put it here because it's a unit test of a function defined in this module and I think tests should be as close to what they're testing as possible. Maybe others have other ideas of how we should organize our tests, and I would love to have that conversation, because I feel our tests are quite disorganized.